api/v1alpha1/operator_types_test.go

```diff
@@ -0,0 +1,83 @@
+package v1alpha1_test
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"strconv"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	operatorutil "github.com/operator-framework/operator-controller/internal/util"
+)
+
+var _ = Describe("OperatorTypes", func() {
+	Describe("Condition Type and Reason constants", func() {
+		It("should register types in operatorutil.ConditionTypes", func() {
+			types, err := parseConstants("Type")
+			Expect(err).NotTo(HaveOccurred())
+
+			for _, t := range types {
+				Expect(t).To(BeElementOf(operatorutil.ConditionTypes), "Append Type%s to operatorutil.ConditionTypes in this package's init function.", t)
+			}
+			for _, t := range operatorutil.ConditionTypes {
+				Expect(t).To(BeElementOf(types), "There must be a Type%[1]s string literal constant for type %[1]q (i.e. 'const Type%[1]s = %[1]q')", t)
+			}
+		})
+		It("should register reasons in operatorutil.ConditionReasons", func() {
+			reasons, err := parseConstants("Reason")
+			Expect(err).NotTo(HaveOccurred())
+
+			for _, r := range reasons {
+				Expect(r).To(BeElementOf(operatorutil.ConditionReasons), "Append Reason%s to operatorutil.ConditionReasons in this package's init function.", r)
+			}
+			for _, r := range operatorutil.ConditionReasons {
+				Expect(r).To(BeElementOf(reasons), "There must be a Reason%[1]s string literal constant for reason %[1]q (i.e. 'const Reason%[1]s = %[1]q')", r)
+			}
+		})
+	})
+})
+
+func parseConstants(prefix string) ([]string, error) {
```

Comment:
nice! could be cool to add a couple of comments for those unfamiliar with the the ast package

Reply:
:man_cook: :kiss: